src/qibolab/qubits.py

```diff
@@ -75,10 +76,25 @@ def __post_init__(self):
 
     @property
     def channels(self):
-        for channel in [self.readout, self.feedback, self.drive, self.flux, self.twpa]:
+        for name in self.CHANNEL_NAMES:
+            channel = getattr(self, name)
```

Comment:
Making it an object?

```py
@dataclass
class Channels:
    readout: Channel
    feedback: Channel
    ...
```

Reply:
Good suggestion in terms of code, the only issue is that from the user perspective the interface `qubit.readout.attenuation` will then be `qubit.channels.readout.attenuation` which is more complicated. Of course this is easy to solve by making
```py
@property
def readout(self):
    return self.channels.readout
```
in the `Qubit`, but then we will have four properties and four setters so the code will end up more complicated than it is now.

Reply:
Wait, since this is a property, isn't it already `qubit.channels.readout.attenuation`?

Reply:
Ok, I see your point. I can make one further proposal, but it has its own trade-offs: we could generate the properties with a decorator.

```py
class Channels:
    @classmethod
    def register(cls, qubit_cls):
        for ch in CHANNELS:
            @property
            def prop(self):
                 return getattr(self.channels, ch)
		setattr(qubit_cls, ch, prop)
        return qubit_cls

@Channels.register()
class Qubit:
    ...
```

The disadvantage is that the code is even more complex (since dynamical). The advantage is that you don't have to repeat anything, and, as in the above proposal, you split part of the complexity out of the `Qubit` class.

To be fair: if I should choose, I'd still go for defining the properties as shortcuts, or leave the user the pleasure of the nesting (I like grouping related things, instead of having giant containers, but this of course generates nesting). However, there are many options, you can choose.

P.S.: it doesn't even have to be so nested in the user code, since you can make intermediate assignments: in the block in which you're working with a qubit, you could alias the channel you need
```py
readout = qubit.channels.readout
do_something(readout.attenuation)
```
but yes, nesting too much is definitely annoying